### PR TITLE
Fix 'Save' FAB icon showing up after updating Townhall settings

### PR DESCRIPTION
**Pull request by JosephPark259**
Not sure if this is a permanent solution but I just use a shallow copy (save on temp variable) and set the state as so.

Let me know what you think

### CODE REVIEW

**Comment by JosephPark259** [comment]
@DevelopmentByDavid Updated!

File: src/contexts/Townhall.tsx [hunk 1]
```diff
@@ -18,6 +18,12 @@ interface Props {
 
 export const TownhallContext = React.createContext<Townhall | null>(null);
 
+type Dispatch = React.Dispatch<SetStateAction<Townhall | undefined >>  | null;
```
Comment (DevelopmentByDavid) [inline src/contexts/Townhall.tsx:21]: I'm not sure that inside the SetStateAction we should have `Townhall | undefined` I think it should just be Townhall and undefined is outside of the generic like the null.

File: src/domains/Townhall/TownhallSettings/TownhallSettings.tsx [hunk 2]
```diff
@@ -50,6 +50,8 @@ export default function TownhallSettings() {
     const [save, isLoading] = useEndpoint(() => configureTownhall(townhall._id, state), {
         onSuccess: () => {
             snack('Saved!');
+            
+            setTownhall({...townhall, settings:state});
```
Comment (DevelopmentByDavid) [inline src/domains/Townhall/TownhallSettings/TownhallSettings.tsx:54]: This should probably use `setTownhall(prev => ({ ...townhall, settings: state}));`. It's a bit hard to explain over text, but I could do a call with you to explain.

**Review by DevelopmentByDavid — CHANGES_REQUESTED** [review]
Looks good!  Two minor comments.

File: src/contexts/Townhall.tsx (hunk 1 above)
Comment (DevelopmentByDavid) [reply src/contexts/Townhall.tsx:21]: I'm wrong!

**Pull request merged**